controllers/nodehealthcheck_controller.go

```diff
@@ -435,7 +436,10 @@ func (r *NodeHealthCheckReconciler) matchesUnhealthyConditions(nhc *remediationv
 			if now.After(n.LastTransitionTime.Add(c.Duration.Duration)) {
 				// unhealthy condition duration expired, node is unhealthy
 				r.Log.Info("Node matches unhealthy condition", "node", node.GetName(), "condition type", c.Type, "condition status", c.Status)
-				commonevents.NormalEventf(r.Recorder, nhc, utils.EventReasonDetectedUnhealthy, "Node matches unhealthy condition. Node %q, condition type %q, condition status %q", node.GetName(), c.Type, c.Status)
+				//In case a remediation is already created no need to repeat this event
+				if !isNodeRemediating {
```

Comment:
we miss events for new unhealthy conditions now

Reply:
The intention was to create the event only for the "first" unhealthy condition, before a CR is created ( `!isNodeRemediating` ) in order to prevent duplicate events.

IIUC we will miss a new unhealthy event condition only when one unhealthy condition was changed to another unhealthy condition.
Another use case we can miss an event is when a status of an already unhealthy condition was changes (and is still considered unhealthy).
I think that both this use-cases qualify as the desirable behavior, let me know if I misunderstood your intention or missed something.

Reply:
that's what I mean 👍🏼